### PR TITLE
docs: Remove table of contents

### PR DESCRIPTION
**Pull request by jodh-intel**
Removed all TOCs now that GitHub auto-generates them.

Also updated the documentation requirements doc removing the requirement
to add a TOC.

Fixes: #2022.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: tools/packaging/release/README.md [hunk 1]
```diff
@@ -34,7 +27,7 @@ $ ./update-repository-version.sh -h
 To update project version for Kata Containers, use the following:
 
 ```bash
-$ make bump-kata-version NEW_VERSION=<new-version>
+$ make bump-kata-version NEW_VERSION="${new_version}"
```
Comment (liubin) [inline tools/packaging/release/README.md:30]: `${new_version}` let image it is a variable, but they are no assignment in this document.

**Comment by liubin** [comment]
Thank you @jodh-intel , removing the toc will save us much time.

File: tools/packaging/release/README.md (hunk 1 above)
Comment (jodh-intel) [reply tools/packaging/release/README.md:32]: Fixed.

**Comment by jodh-intel** [comment]
Hmm, now that https://github.com/kata-containers/tests/pull/3625 has landed, looking at the static checks failure we see lots of HTTP 429 errors, meaning we've hit the rate limiter for GitHub:

```
ERROR: found HTTP error status codes for URL https://github.com/kubernetes/minikube/blob/master/deploy/addons/efk/fluentd-es-configmap.yaml.tmpl (429)
ERROR: found HTTP error status codes for URL https://github.com/kubernetes/minikube/blob/master/deploy/iso/minikube-iso/board/coreos/minikube/rootfs-overlay/etc/systemd/journald.conf (429)
```

I assume that limit is due to the unauthenticated connections we're using to check these URLs. Newer versions of `curl` are supposed to honour the `Retry-After:` value in the response so I'm wondering if either we're using an older version of `curl`, or possibly that all the coprocesses running `curl` do the backoff thing, then all try to hit the site again, resulting in 429 again.

I think we can force-merge this PR, but we do need a solution going forward (although admittedly, not many PRs touch so many documents so the chance of hitting this may in fact be quite low). I guess we could use a secret and pass creds to `curl` so that it can authenticate as a generic user maybe?

**Review by liubin — APPROVED** [review]

**Comment by jodh-intel** [comment]
We need one more ack here so we can land these changes ;)

**Review by devimc — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -126,9 +110,9 @@ The following repositories are used by both the current and first generation Kat
 
 | Component | Description | Current | First generation | Notes |
 |-|-|-|-|-|
-| CI | Continuous Integration configuration files and scripts. | [Kata 2.x](https://github.com/kata-containers/ci/tree/main) | [Kata 1.x](https://github.com/kata-containers/ci/tree/master) | |
+| CI | Continuous Integration configuration files and scripts. | [Kata 2.x](https://github.com/kata-containers/ci/tree/master) | [Kata 1.x](https://github.com/kata-containers/ci/tree/master) | |
```
Comment (fidencio) [inline README.md:113]: Hmm. This is not consistent.

We should move to using `main` on all our repos, for the sake of consistency.
Also, we should not keep references of 1.x, as this is gone for good at this point.

File: README.md [hunk 3]
```diff
@@ -126,9 +110,9 @@ The following repositories are used by both the current and first generation Kat
...
 | kernel | The Linux kernel used by the hypervisor to boot the guest image. | [Kata 2.x][kernel] | [Kata 1.x][kernel] | Patches are stored in the packaging component. |
-| tests | Test code. | [Kata 2.x](https://github.com/kata-containers/tests/tree/main) | [Kata 1.x](https://github.com/kata-containers/tests/tree/master) | Excludes unit tests which live with the main code. |
+| tests | Test code. | [Kata 2.x](https://github.com/kata-containers/tests/tree/main) | [Kata 1.x](https://github.com/kata-containers/tests/tree/main) | Excludes unit tests which live with the main code. |
```
Comment (fidencio) [inline README.md:115]: Ditto.

**Review by fidencio — COMMENTED** [review]
@jodh-intel, I've left some comments, let me know what do you think.

File: README.md (hunk 2 above)
Comment (jodh-intel) [reply README.md]: This is kinda out of scope of this PR, but I tend to agree, so I've added an extra commit to rip out the 1.x stuff ;)

**Comment by jodh-intel** [comment]
The "test" CI's are blocking this PR from landing ;(

**Comment by fidencio** [comment]
@jodh-intel, sorry I tried closing and re-opening as it was a known workaround but that didn't work.
I've fixed the issue this morning (see: https://github.com/kata-containers/kata-containers/commit/5957bc7d9cd252937744fb658dbd83422e666055), would be possible to rebase your PR atop of `main`?

**Comment by jodh-intel** [comment]
Rebase. Let's see if we can land this PR this month... ;)

/test

**Comment by fidencio** [comment]
just for the record, the static-checks were "stuck" due to the `force-skip-ci` label.  So, what we should be doing, IMHO, is short-circuiting and returning success on the static checks when the `force-skip-ci` label is present, rather than not running those.

**Comment by jodh-intel** [comment]
\o/ No words, just tears of happiness! :joy:

**Pull request merged**